src/main/java/net/minestom/server/entity/ai/goal/RandomStrollGoal.java

```diff
@@ -32,17 +43,26 @@ public boolean shouldStart() {
 
     @Override
     public void start() {
-        int remainingAttempt = closePositions.size();
-        while (remainingAttempt-- > 0) {
-            final int index = random.nextInt(closePositions.size());
-            final Vec position = closePositions.get(index);
-
-            final var target = entityCreature.getPosition().add(position);
-            final boolean result = entityCreature.getNavigator().setPathTo(target);
-            if (result) {
-                break;
+        final int remainingAttempt = closePositions.size();
+        CompletableFuture.runAsync(() -> {
```

Comment:
Got a few things I want to say here:
1. Shouldn't the future use an existing execution pool? Because otherwise, it'll use `ForkJoinPool.commonPool()`, which may not be desired, as we already have existing scheduling pools.
2. Couldn't this be a scheduled task using the scheduler so it can be tracked more easily?
3. If we are going to stick with futures for these things, I think the method should return the future, so that users can call back to it.
4. I think it might even be better if the goals weren't the ones handling the concurrency at all, and instead that was left to the thing managing all of the AI processing.
5. Why is this the only goal that uses a future to compute the logic? Am I missing something here?

I may be talking out of my ass here, idk, but I think these are valid points.